### ISSUE TITLE
animateAnchor:false not working when using scrollBar:true

### ISSUE THREAD

**Issue by Annihilus** [alvarotrigo/fullPage.js#2584]
Standard behavior of the browser, when you refresh or back to a page includes the preservation of scroll position. Fullpage also always returns to the first section and scrolls the current section.Can I abandon this behavior?

**Comment by alvarotrigo** [comment]
> Can I abandon this behavior?

Do not use the `anchors` option.

Or, if you are referring to the animaton, use `animateAnchor:false`.

**Comment by Annihilus** [comment]
If I turn off anchor option  when I reload the page Im still stay on the first slide 
But in this situation there is not scrolling to the slide
So after the reloading of the page I want to be on that exact place where I was before the refreshing(reloading) of the page .
animateAnchor didn't help me, it scrolles to the section only when I'm using the scroll

**Comment by alvarotrigo** [comment]
> Animateanchor transfers it on slide but it didn't work if I use scrollBar:true

Working as expected on the [scrollBar demo page](http://alvarotrigo.com/fullPage/examples/scrollBar.html#secondPage/1)

**Comment by Annihilus** [comment]
This result is not that I expected
I don't need animation, which starts with the first section and goes to the current section.
I want to begin with the current section after reloading of the page.

**Comment by alvarotrigo** [comment]
Right. Then `animateAnchor:false` doesn't seem to be working when using `scrollBar:true`.
I'll mark it as a bug to investigate.

**Comment by rrthomas** [comment]
I've done a deep dive on this in the last couple of days. I've been primarily working in Firefox, but Chrome has the same problem, though it often works there. This is not surprising, because it seems to be a race condition.

The principal problem is that on reload the browser scrolls back up to the top of the page. I don't know why it does this; maybe because there is no actual anchor corresponding to the URL? But fullPage responds by updating the URL to point to successively earlier sections until it gets to the top. You can see some of the slide numbers in browser history; conversely, by following along with tracing in dev tools, the only calls to fullPage code are coming from the scroll events, so it is definitely the browser that is causing the scroll.

I believe it's a race condition because if the scroll back to the start happens fast enough then fullPage does not change the URL and the eventual call to `scrollToAnchor` still has the correct destination to go to.

I have a fix. I will not be surprised to hear that it is not a good fix!

The fix comes in two parts:

1. Call to `scrollToAnchor` on the `onload` event, not the `DOMContentLoaded` event. This waits until after the browser has finished scrolling the page backwards.
2. Move the calls to bind `scrollHandler` to scroll events from `bindEvents$1` to `scrollToAnchor`. This prevents fullPage from updating the URL while the browser is scrolling the window.

The complete patch to `fullpage.js` (non-minified, of course) looks like this for current `master`:

```js
--- /home/rrt/repo/fullPage.js/dist/fullpage.js 2026-08-24 20:16:15.196935612 +0100
+++ fullpage.js 2026-08-24 22:12:08.135623725 +0100
@@ -5328,8 +5328,6 @@
       windowRemoveEvent('scroll', scrollHandler);
     }
     function bindEvents$1() {
-      windowAddEvent('scroll', scrollHandler);
-      doc.body.addEventListener('scroll', scrollHandler);
       EventEmitter.on(events.onScrollPageAndSlide, function (params) {
         scrollPageAndSlide(params.sectionAnchor, params.slideAnchor);
       });
@@ -5638,6 +5636,8 @@
     * Scrolls to the anchor in the URL when loading the site
     */
     function scrollToAnchor() {
+      windowAddEvent('scroll', scrollHandler);
+      doc.body.addEventListener('scroll', scrollHandler);
       var anchors = getAnchorsURL();
       var sectionAnchor = anchors.section;
       var slideAnchor = anchors.slide;
@@ -5768,7 +5768,7 @@
       if (document.readyState !== 'loading') {
         scrollToAnchor();
       } else {
-        document.addEventListener('DOMContentLoaded', scrollToAnchor);
+        window.addEventListener('load', scrollToAnchor);
       }
       afterRenderActions();
 
```

Just to emphasise, I'm not saying that this patch is a good idea (hence I'm not proposing it as a PR), but hopefully it illustrates the problem more concretely.

**Comment by alvarotrigo** [comment]
> Call to scrollToAnchor on the onload event, not the DOMContentLoaded event. This waits until after the browser has finished scrolling the page backwards.

The load event was replaced for domready to fix the fact scrolling to sections wouldn't work as expected when using large media contents. Basically, the page didnt scroll to the section until every single element was loaded.
See [this issue](https://github.com/alvarotrigo/fullPage.js/issues/4711) as a reference.

> Move the calls to bind scrollHandler to scroll events from bindEvents$1 to scrollToAnchor. This prevents fullPage from updating the URL while the browser is scrolling the window.

I'll try this to see if that works. Thanks! ;)
